test/powershell/engine/ETS/Adapter.Tests.ps1

```diff
@@ -80,3 +80,24 @@ Describe "Adapter Tests" -tags "CI" {
         $val | Should Be "book"
     }
 }
+
+Describe "Adapter XML Tests" -tags "CI" {
+    BeforeAll {
+        [xml]$x  = "<root><data/></root>"
+        $testCases =
+            @{ rval = @{testprop = 1}; value = 'hash (psobject)' },
+            @{ rval = 1;               value = 'int (codemethod)' },
+            @{ rval = "teststring";    value = 'string (codemethod)' }
+    }

```

Comment:
I would add tests for:
* $null
* array of strings
* array of integers
* psobject wrapping each of the previous values